raster-foundry/raster-foundry
# Add AOI create UI and component

## Overview

~**Added WIP tag to fix the known bugs**~

This PR adds AOI creation UI and a reusable component with a secondary bar that has map draw and edit handlers along with confirm and cancel buttons, so that, together, they enable:
- draw, edit, delete AOIs
- view, edit, apply existing AOIs of other layers in the same project, or AOIs stored in user's vector data
- save AOI as vector data for later use in the app

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo

1. Create an AOI

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/create.gif)

2. Modify an AOI

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/modify.gif)

3. Redraw and AOI

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/redraw.gif)

3. Choose an AOI from another layer

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/choose-from-layer.gif)

4. Choose an AOI from vector data

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/choose-from-vector-data.gif)

5. Save AOI as vector data

![video-play-close](https://raw.githubusercontent.com/aaronxsu/MyData/master/gifs/rf-pr-gifs/save-as-vector.gif)

### Notes

~There are two know minor bugs:~
~1. When choosing vector data created from v1 scene browse UI's shape creation, it cannot find geometry~
~2. Directly applying an existing layer's AOI or an existing shape from vector data is not allowed~

~I will fix the above two in the morning.~

^^ Done

## Testing Instructions

 * Create a new layer in v2 project UI
 * Click on the warning sign and go to AOI creation UI
 * Use the UI to create, edit, delete, select AOIs -- make sure they work as you expect

Closes https://github.com/raster-foundry/raster-foundry/issues/4574
